share/availability/light/availability_test.go

```diff
@@ -359,31 +351,23 @@ func TestPruneAll(t *testing.T) {
 
 func TestPrunePartialFailed(t *testing.T) {
 	const size = 8
-	ctx, cancel := context.WithTimeout(context.Background(), time.Second*2)
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*200)
 	t.Cleanup(cancel)
 
-	dir := t.TempDir()
-	store, err := store.NewStore(store.DefaultParameters(), dir)
-	require.NoError(t, err)
-	defer require.NoError(t, store.Stop(ctx))
 	eds, h := randEdsAndHeader(t, size)
-	err = store.PutODSQ4(ctx, h.DAH, h.Height(), eds)
-	require.NoError(t, err)
-
-	// Create a new bitswap getter
 	ds := ds_sync.MutexWrap(datastore.NewMapDatastore())
 	clientBs := blockstore.NewBlockstore(ds)
-	serverBS := newHalfFailBlockstore(&bitswap.Blockstore{Getter: store})
-	ex := newFakeExchange(serverBS)
+
+	ex := newHalfSessionExchange(newExchangeOverEDS(ctx, t, eds))
 	getter := bitswap.NewGetter(ex, clientBs, 0)
 	getter.Start()
 	defer getter.Stop()
 
 	// Create a new ShareAvailability instance and sample the shares
 	sampleAmount := uint(20)
 	avail := NewShareAvailability(getter, ds, clientBs, WithSampleAmount(sampleAmount))
-	err = avail.SharesAvailable(ctx, h)
-	require.NoError(t, err)
+	err := avail.SharesAvailable(ctx, h)
+	require.Error(t, err)
```

Comment:
This must have been error, because we fail half of requests and thus the error comes up